astropy/table/table.py

```diff
@@ -463,8 +463,7 @@ def __init__(self, data=None, masked=False, names=None, dtype=None,
             elif isinstance(rows, self.Row):
                 data = rows
             else:
-                rec_data = recarray_fromrecords(rows)
```

Comment:
I just realized this removes the only use of `recarray_fromrecords`, so we might as well remove it from `np_utils.py` - it only means this PR fixes #8976 (will add that on top too).

Reply:
Fixed in 7468364.